components/Typography/tests/unit/MaterialScalableFontTests.m

```diff
@@ -0,0 +1,266 @@
+    UIContentSizeCategoryAccessibilityExtraLarge,
+    UIContentSizeCategoryAccessibilityExtraExtraLarge,
+    UIContentSizeCategoryAccessibilityExtraExtraExtraLarge,
+  ];
+
+  for (MDCTextStyle textStyle in textStyles) {
+    // When
+    UIFont *font = [UIFont systemFontOfSize:18.0];
+
+    MDCFontScaler *scaler = [[MDCFontScaler alloc] initForMaterialTextStyle:textStyle];
+
+    UIFont *scalableFont = [scaler scalableFontWithFont:font];
+
+    for (unsigned long ii = 0; ii < sizeCategories.count - 1; ++ii) {
+      UIContentSizeCategory smallerSizeCategory = sizeCategories[ii];
+      UIContentSizeCategory largerSizeCategory = sizeCategories[ii + 1];
+
+      UIFont *smallerFont = [scalableFont mdc_scaledFontForSizeCategory:smallerSizeCategory];
+      UIFont *largerFont = [scalableFont mdc_scaledFontForSizeCategory:largerSizeCategory];
+
+      // Then
+      XCTAssert(smallerFont.pointSize <= largerFont.pointSize);
+    }
+  }
+}
+
+- (void)testScaledFontDefaultEqualsLarge {
+  // Given
+  NSArray<MDCTextStyle> *textStyles = @[
+    MDCTextStyleHeadline1,
+    MDCTextStyleHeadline2,
+    MDCTextStyleHeadline3,
+    MDCTextStyleHeadline4,
+    MDCTextStyleHeadline5,
+    MDCTextStyleHeadline6,
+    MDCTextStyleSubtitle1,
+    MDCTextStyleSubtitle2,
+    MDCTextStyleBody1,
+    MDCTextStyleBody2,
+    MDCTextStyleButton,
+    MDCTextStyleCaption,
+    MDCTextStyleOverline,
+  ];
+
+  for (MDCTextStyle textStyle in textStyles) {
+    // When
+    UIFont *font = [UIFont systemFontOfSize:18.0];
+
+    MDCFontScaler *scaler = [[MDCFontScaler alloc] initForMaterialTextStyle:textStyle];
+
+    UIFont *scalabledFont = [scaler scalableFontWithFont:font];
+
+    UIFont *defaultFont = [scalabledFont mdc_scaledFontAtDefaultSize];
+    UIFont *largeFont = [scalabledFont mdc_scaledFontForSizeCategory:UIContentSizeCategoryLarge];
+
+    // Then
+    XCTAssert([defaultFont mdc_isSimplyEqual:largeFont]);
+  }
+}
+
+- (void)testInvalidStyleFallback {
+  // Given
+  UIFont *font = [UIFont systemFontOfSize:18.0];
+
+  MDCFontScaler *scaler1 = [[MDCFontScaler alloc] initForMaterialTextStyle:MDCTextStyleBody1];
+  MDCFontScaler *scaler2 =
+      [[MDCFontScaler alloc] initForMaterialTextStyle:@"IntentionallyNonTextStyleString"];
+
+  UIFont *scalableFont1 = [scaler1 scalableFontWithFont:font];
+  UIFont *scalableFont2 = [scaler2 scalableFontWithFont:font];
+
+  UIFont *scaledFont1 = [scalableFont1 mdc_scaledFontAtDefaultSize];
+  UIFont *scaledFont2 = [scalableFont2 mdc_scaledFontAtDefaultSize];
+
+  // Then
+  XCTAssert([scaledFont1 mdc_isSimplyEqual:scaledFont2]);
+}
+
+- (void)testNonScaledFontReturnsSelf {
+  // Given
+  UIFont *font = [UIFont systemFontOfSize:18.0];
+
+  // Note that no scaling curve has been attached to the font, so it will NOT scale
+  UIFont *nonScaledFont1 = [font mdc_scaledFontForSizeCategory:UIContentSizeCategoryExtraLarge];
+
+  // Then
+  XCTAssert([font mdc_isSimplyEqual:nonScaledFont1]);
```

Comment:
Please `XCTAssertEqual` to ensure it returns self. At the moment, your test is only asserting that some properties are equivalent.

Reply:
Testing font equality is non-trivial :/
UIFont.mdc_isSimplyEqual was created to work around some of those issues.

Reply:
Ack.